specification/quantumleap.yml

```diff
@@ -0,0 +1,500 @@
+      summary: "History of an attribute (values only) of a given entity instance."
+      description: "Similar to the previous, but focusing on the values regardles of the metadata."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        - $ref: '#/parameters/attrNameInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
+
+
+  /entities/{entityId}:
+    get:
+      operationId: reporter.reporter.query_1T1ENA
+      summary: "History of N attributes of a given entity instance."
+      description: "For example, query the average pressure, temperature and humidity of this month in the weather station WS1."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+        - $ref: '#/parameters/attrs'
+        - $ref: '#/parameters/aggrMethod'
+        - $ref: '#/parameters/aggrPeriod'
+        - $ref: '#/parameters/options'
+        - $ref: '#/parameters/from'
+        - $ref: '#/parameters/to'
+        - $ref: '#/parameters/lastN'
+        - $ref: '#/parameters/limit'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+              attrName:
+                type: string
+              index:
+                type: array
+                items:
+                  type: number
+                description: "Array of the timestamps of the requested data. Values are UTC milliseconds since 1970-01-01T00:00:00Z."
+              values:
+                type: array
+                items:
+                  type: number
+                description: "Array of values of the selected attribute, in the same corresponding order of the 'index' array."
+          examples:
+            application/json:
+              {
+                "id": "Kitchen1",
+                "attrName": "temperature",
+                "index": [1510826981388, 1510826990438, 1510826999725],
+                "values": [24.1, 25.3, 26.7]
+              }
+
+  /entities/{entityId}/value:
+    get:
+      operationId: reporter.reporter.query_1T1ENA_value
+      summary: "History of N attributes (values only) of a given entity instance."
+      description: "For example, query the average pressure, temperature and humidity (values only, no metadata) of this month in the weather station WS1."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+        - $ref: '#/parameters/attrs'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
```

Comment:
a good approach (are array of arrays possible in json?) could be (and when for a time index there is not value for an attribute, we set it to null):

```
{
     "index": [1510826981388, 1510826990438, 1510826999725],
     "values": [
          [24.1, 25.3, 26.7],
          [1.01, 0.9, 1.02]
     ]
}
```

Reply:
Arrays of arrays are valid AFAIK, but we'll probably need a "map" there to tell which array corresponds to which attribute (because I don't recall reading about default explicit attributes order)

Reply:
can't it be a semantic we define, for example based on the order of attributes in the query?

?attrs=temperature,pressure&...